# Occupation and Skill-Level Analysis

Compare annual changes in occupation and skill-level advertisement
indices. Detailed skill categories and broad skill groupings are
analysed separately. Index values are not vacancy counts.

In [1]:
from pathlib import Path

import pandas as pd

processed_dir = Path("../data/processed")

analysis_files = {
    "Occupation": "occupation_quarterly.csv",
    "Detailed skill": "skill_levels_quarterly.csv",
    "Broad skill and overall": "broad_skill_quarterly.csv"
}

analysis_tables = {}

for name, filename in analysis_files.items():
    table = pd.read_csv(
        processed_dir / filename,
        parse_dates=["Date"],
        index_col="Date"
    )

    analysis_tables[name] = table

In [3]:
annual_growth_tables = {}

for name, table in analysis_tables.items():
    growth = table.pct_change(
        periods=4,
        fill_method=None
    ) * 100

    annual_growth_tables[name] = growth

    print(name, "—", growth.index[-1].to_period("Q"))
    display(growth.iloc[-1].sort_values(ascending=False).round(2))
    print("------------------------")

Occupation — 2026Q2


Machinery Drivers                19.84
Sales                            15.21
Labourers                         9.29
Tech & Trades                     7.68
Clerical & Administration         7.35
Professionals                     4.94
Managers                          4.58
Community & Personal Services    -0.18
Name: 2026-06-01 00:00:00, dtype: float64

------------------------
Detailed skill — 2026Q2


unskilled         12.12
low skilled        9.41
semi-skilled       9.16
highly skilled     6.85
skilled           -0.30
Name: 2026-06-01 00:00:00, dtype: float64

------------------------
Broad skill and overall — 2026Q2


Unskilled    10.21
ALL           6.72
Skilled       5.51
Name: 2026-06-01 00:00:00, dtype: float64

------------------------


In [4]:
latest_broad_growth = (
    annual_growth_tables["Broad skill and overall"].iloc[-1]
)

broad_comparison = (
    latest_broad_growth[["Skilled", "Unskilled"]]
    .rename("yoy_change_pct")
    .to_frame()
)

broad_comparison["difference_from_overall_pp"] = (
    broad_comparison["yoy_change_pct"]
    - latest_broad_growth["ALL"]
)

broad_comparison.round(2)

,yoy_change_pct,difference_from_overall_pp
Skilled,5.51,-1.21
Unskilled,10.21,3.49


### Conclusion: Occupation and skill-level growth

In 2026Q2, seven of eight occupation groups recorded positive annual
growth. Machinery Drivers led at 19.84%, while Community & Personal
Services declined by 0.18%.

Four of five detailed skill categories grew. The `unskilled` category
led at 12.12%, while the detailed `skilled` category declined by 0.30%.

Broad `Unskilled` growth was 10.21%, exceeding overall growth of 6.72%
by 3.49 percentage points. Broad `Skilled` grew by 5.51%.

Detailed categories and broad groupings are not interchangeable.
These results measure changes in advertisement indices, not
advertisement counts or employment shares.

### Classification reference

MBIE's 2018 Jobs Online methodology groups highly skilled, skilled,
and semi-skilled categories into broad `Skilled`, and low skilled
and unskilled categories into broad `Unskilled`.

[MBIE Jobs Online methodology (2018)](https://www.mbie.govt.nz/assets/82f9a170cc/jobs-online-methodology-2018.pdf)